src/coreclr/nativeaot/System.Private.CoreLib/src/System/InvokeUtils.cs

```diff
@@ -215,14 +215,20 @@ private static Exception ConvertOrWidenPrimitivesEnumsAndPointersIfPossible(obje
             return null;
         }
 
-        private static Exception ConvertPointerIfPossible(object srcObject, EETypePtr dstEEType, CheckArgumentSemantics semantics, out IntPtr dstIntPtr)
+        private static Exception ConvertPointerIfPossible(object srcObject, EETypePtr dstEEType, CheckArgumentSemantics semantics, out object dstIntPtr)
         {
             if (srcObject is IntPtr srcIntPtr)
             {
                 dstIntPtr = srcIntPtr;
                 return null;
             }
 
+            if (srcObject is UIntPtr srcUIntPtr)
+            {
+                dstIntPtr = srcUIntPtr;
+                return null;
+            }
+
             if (srcObject is Pointer srcPointer)
             {
                 if (dstEEType == typeof(void*).TypeHandle.ToEETypePtr() || RuntimeImports.AreTypesAssignable(pSourceType: srcPointer.GetPointerType().TypeHandle.ToEETypePtr(), pTargetType: dstEEType))
```

Comment:
Also change `dstIntPtr = IntPtr.Zero;` to `dstPtr = null` below. We do not need to create IntPtr.Zero box since we are going to throw exception anyway.